### ISSUE TITLE
Social accounts: login and sign up account sorting

### ISSUE THREAD

**Issue by humitos** [readthedocs/ext-theme#379]
I found myself wanting a different sorting for Social Account providers that the one we are exposing:

![Screenshot_2024-06-12_16-19-43](https://github.com/readthedocs/ext-theme/assets/244656/f2aefa8d-24f2-4563-aa5c-9f8c0229b475)

Ideally, I'd put GitHub and Google at the top and then GitLab and Bitbucket which are less used.

1. GitHub
2. Google
3. GitLab
4. Bitbucket

They are currently sorted by name, but I'd change that sorting in a way we can promote more GitHub and Google: https://github.com/pennersr/django-allauth/blob/38eb3114ab6c1b63359e51407a7349606bb50740/allauth/socialaccount/templatetags/socialaccount.py#L86

Bitbucket in particular, it the first one while it's the least used.

**Comment by agjohnson** [comment]
That sorting feels unintentional to me. Agreed about demoting Bitbucket, but with Google being it's own thing and not a repository provider, I'd still keep that at the bottom: GitHub, GitLab, Bitbucket, Google.

It would be nice if we could do this without changing the Allauth usage, but I don't see an easy way of doing that. This probably requires a one-off template filter after `get_providers`.

**Comment by humitos** [comment]
I would prioritize UX over the specific thing of Google not being VCS. GitHub and Google are the ones we want to promote more and the ones that users use more, so they should be the easiest ones to access.

However, we are want to keep the distinction of VCS and Google, we can make Google to be on its own column:

![339008365-f2aefa8d-24f2-4563-aa5c-9f8c0229b475](https://github.com/user-attachments/assets/47238461-c790-46c7-8b4a-397d64e25e16)

This way, we are differentiating Google from the rest and promoting GitHub at the same time.

**Comment by agjohnson** [comment]
I don't think that it's necessary, Google isn't exactly a better experience compared to the others -- and it feels more prominent separated. Other providers just group all of their buttons together and I think that is fine. I don't want to complicate this UI too much more than it already is given the number of authentication methods we have.

**Comment by humitos** [comment]
I'm not saying that Google is _better_. I'm just differentiating it from VCS because you mentioned it would be weird to list it between the other VCS providers.

My goal here is to make GitHub and Google _easier_ to click for users since they are the most used ones in our platform and the ones we want our users to use the most. GitLab and Bitbucket have less features in our platform.

If you consider my suggestion to reach that goal could be better, what would you change to it?
